pkg/router/controller/factory/factory.go

```diff
@@ -56,25 +57,39 @@ func NewDefaultRouterControllerFactory(oc osclient.RoutesNamespacer, kc kclients
 	}
 }
 
+func routerKeyFn(obj interface{}) (string, error) {
```

Comment:
Can you just put a comment in here explaining where this came from and why we pulled it in here.